### PR TITLE
[azure-blobstorage-artifactProvider-v2] Modified 'get' method to save file from Azure Storage correctly 

### PR DESCRIPTION
**Pull request by egor-bryzgalov**
Related issue: #10683 

**azure-blobstorage-artifactProvider-v2**:
- Added code replacing ArtifactItem object full path by a filename after the read stream is created to make sure the file is saved properly to target folder;
- Fixed issue with missing loc strings for the task's logging.

**JavaToolInstallerV0**:
- Updated dependencies (azure-blobstorage-artifactProvider replaced by it's second version).
- Updated resources.resjson as addition to PR #12929.

### CODE REVIEW

File: Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts [hunk 1]
```diff
@@ -155,6 +162,17 @@ export class AzureBlobProvider implements models.IArtifactProvider {
         return artifactItems;
     }
 
+     /**
+     * Get file name from full path to this file.
+     * @returns {string} File name without folders
+     * @param {string} path Path to file
+     */
+    private _getFileName(path: string): string {
```
Comment (AlexandrKravchuk) [inline Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts:170]: Might be static

Comment (AlexandrKravchuk) [reply Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts:170]: Seems like it's possible to use [path.basename](https://stackoverflow.com/questions/19811541/nodejs-get-file-name-from-absolute-path) instead?

File: Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts [hunk 2]
```diff
@@ -73,6 +76,10 @@ export class AzureBlobProvider implements models.IArtifactProvider {
             } else {
                 readStream = this._blobSvc.createReadStream(this._container, artifactItem.path, null);
             }
+            const fileName: string = this._getFileName(artifactItem.path);
+            if (fileName) {
+                artifactItem.path = fileName;
```
Comment (AlexandrKravchuk) [inline Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts:81]: artifactItem.path = this._getFileName(artifactItem.path) || artifactItem.path
As an idea for shortening

File: Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts (hunk 1 above)
Comment (egor-bryzgalov) [reply Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts:170]: > Seems like it's possible to use [path.basename](https://stackoverflow.com/questions/19811541/nodejs-get-file-name-from-absolute-path) instead?

Thank you for your suggestion! It makes sense.

File: Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts (hunk 2 above)
Comment (egor-bryzgalov) [reply Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts]: @AlexandrKravchuk these lines are removed in accordance with your previous review point.

File: Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts [hunk 3]
```diff
@@ -154,7 +159,7 @@ export class AzureBlobProvider implements models.IArtifactProvider {
 
         return artifactItems;
     }
-
+
```
Comment (AlexandrKravchuk) [inline Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts:162]: Nit: Redundant change

**Pull request merged**